apps/gdalalg_raster_resize.cpp

```diff
@@ -0,0 +1,102 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  "resize" step of "raster pipeline"
+ * Author:   Even Rouault <even dot rouault at spatialys.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Even Rouault <even dot rouault at spatialys.com>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_raster_resize.h"
+
+#include "gdal_priv.h"
+#include "gdal_utils.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*        GDALRasterResizeAlgorithm::GDALRasterResizeAlgorithm()        */
+/************************************************************************/
+
+GDALRasterResizeAlgorithm::GDALRasterResizeAlgorithm(bool standaloneStep)
+    : GDALRasterPipelineStepAlgorithm(NAME, DESCRIPTION, HELP_URL,
+                                      standaloneStep)
+{
+    auto &sizeArg = AddArg("size", 0, _("Target size in pixels"), &m_size)
+                        .SetMinCount(2)
+                        .SetMaxCount(2)
+                        .SetRequired()
+                        .SetRepeatedArgAllowed(false)
+                        .SetDisplayHintAboutRepetition(false)
+                        .SetMetaVar("<width>,<height>")
+                        .SetMutualExclusionGroup("resolution-size");
+    sizeArg.AddValidationAction(
+        [&sizeArg]()
+        {
+            const auto &val = sizeArg.Get<std::vector<int>>();
+            CPLAssert(val.size() == 2);
+            if (!(val[0] >= 0 && val[1] >= 0))
+            {
+                CPLError(CE_Failure, CPLE_AppDefined,
+                         "Target size should be positive or 0.");
+                return false;
```

Comment:
Is is checked that the target size is in integers, or would the value be rounded or truncated automatically?

Reply:
```
$ gdal raster resize --size=10.5,10 byte.tif out.tif  --overwrite
ERROR 5: resize: Expected list of integer value for argument '--size', but got '10.5,10'.
Usage: gdal raster resize [OPTIONS] <INPUT> <OUTPUT>
Try 'gdal raster resize --help' for help.
```